server/src/main/java/org/opensearch/discovery/InitializeExtensionsRequest.java

```diff
@@ -29,23 +29,32 @@ public class InitializeExtensionsRequest extends TransportRequest {
      * TODO change DiscoveryNode to Extension information
      */
     private final List<DiscoveryExtension> extensions;
+    private static final int OS_DEFAULT_PORT = 9200;
+    private int port;
+    private String hostAddress;
```

Comment:
Since this address is just part of the `sourceNode` why do we need to have an extra field for it? It just creates redundancy in all the reading, writing, equals, hashcode, etc.  You can still use a convenience getter that returns `sourceNode.getHostAddress()`.

Reply:
That's a good point. Let me make that change